### DISCUSSION TITLE
Feature request: hkp search for keys

### DISCUSSION THREAD

**Discussion by kellerben** [openpgpjs/openpgpjs#1207] [Ideas]
It would be nice if one could search for keys. I.e. if one does not know the exact key and needs to display a choice to the user:
e.g.: https://pgp.mit.edu/pks/lookup?search=richard+stallman&op=index

hkp.lookup({query: 'Richard Stallman'}).then(function(array_of_key){
  … ask user which one to use …
})

**Comment by tanx** [comment]
Agreed. The client parser for (v)index should be pretty strait forward. Here is the corresponding server side 

https://github.com/mailvelope/keyserver/blob/master/src/route/hkp.js#L128-L138

I would not put it behind the `query` param though. But rather add a new `index` param. Patches welcome :)

**Comment by kellerben** [comment]
I think it should be implemented in a new function instead of another magic parameter. The return value of this function is totally different to the other function which fetches only one key. My proposal is attached. Feel free to adopt it.
